src/platform/datapath_winuser.c

```diff
@@ -139,6 +139,11 @@ typedef struct CXPLAT_DATAPATH_INTERNAL_RECV_CONTEXT {
     //
     CXPLAT_POOL* OwningPool;
 
+    //
+    // The owning per-processor socket.
+    //
+    CXPLAT_SOCKET_PROC* SocketProc;
```

Comment:
We can eliminate/replace `OwningPool`. It will be `SocketProc->DatapathProc->RecvDatagramPool`.

Reply:
I'd rather leave this because RIO will need the field.

Reply:
There will be multiple possible pools that you can't figure out which to free back to with RIO?

Reply:
(For RIO we'll have a separate pool of registered buffers.)